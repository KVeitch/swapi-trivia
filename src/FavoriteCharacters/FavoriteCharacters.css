.fav__character__btn {
  height:70px;
}

.favorite__h2 {
  font-family: 'SpaceMeatballRegular';
  font-size: 3em;
  font-style: normal;
  font-weight: normal;
  text-shadow: 4px 3px 0px #fff, 9px 8px 0px rgba(0,0,0,0.15);
  margin:0 auto;
}

.favorite__h3 {
  color:rgb(194, 0, 0);
  font-family: 'SpaceMeatballRegular';
  font-size: 2em;
  font-style: normal;
  font-weight: normal;
  text-shadow: 4px 3px 0px #fff, 9px 8px 0px rgba(0,0,0,0.15);
  margin:0 auto;
}