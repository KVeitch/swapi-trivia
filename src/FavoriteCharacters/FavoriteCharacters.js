import React from 'react';
import { Link } from 'react-router-dom'
import CharacterCard from '../CharacterCard/CharacterCard';
import PropTypes from 'prop-types';
import loading from '../images/loading.gif'
import './FavoriteCharacters.css'

const FavoriteCharacters = props => {
    if(props.favoriteList.length===0){
        return(
            <>
            <h2 className='favorite__h2'>See your favorite character!</h2>
            <h3 className='favorite__h3'>Please select a few first.</h3>
            <Link to='/movies'>
                <button className='user__buttons fav__character__btn'>Back to Movies</button>
            </Link>
            <div className='div__image'>
                <img className='loading__image' src={loading} alt="BB-8 droid rolling back and forth" />
            </div>
            </>
        )
    }

const characterList = props.characters.map((character, i) => {

    return <CharacterCard 
                key={'fav'+i} 
                character={character}
                setFavorite={props.setFavorite}
                favoriteList={props.favoriteList} />
    })
    return (
        <section className='character__container'>
            {characterList}
        </section>
    )
}

export default FavoriteCharacters;

FavoriteCharacters.propTypes = {
    characters:PropTypes.array,
    changeSelectedMovie:PropTypes.func,
    isReady:PropTypes.bool,
    setFavorite:PropTypes.func,
    favoriteList:PropTypes.array,
  }