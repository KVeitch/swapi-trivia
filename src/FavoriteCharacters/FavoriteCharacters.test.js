import React from 'react';
import { shallow } from 'enzyme';
import FavoriteCharacters from './FavoriteCharacters';

describe('Favorite Characters', () => {

    it.skip('should match the snapshot with all the data passed in correctly', () => {
        const wrapper =
            shallow(<FavoriteCharacters
                    characters={[]}/>)
        expect(wrapper).toMatchSnapshot();
    });
});