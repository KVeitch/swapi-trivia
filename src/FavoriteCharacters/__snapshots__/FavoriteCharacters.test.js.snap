// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorite Characters should match the snapshot with all the data passed in correctly 1`] = `
<section
  className="character__container"
/>
`;
